API/Array.hpp.in

```diff
@@ -0,0 +1,29 @@
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```

Comment:
generally speaking, classes defined by `typedef` or `using` statement can live in namespace header, but to err on the side of too fine a granularity is better than to err on the side of too coarse a granularity